### DISCUSSION TITLE
PieChartModel primefaces 13.0.0

### DISCUSSION THREAD

**Discussion by zapeSpawn** [primefaces/community#73] [PrimeFaces] [answered]
Hello,
 
With primefaces version 13.0.0, I have a ChartPieModel and I want to be able to show a different content to the value in the tooltip of each option. 
For example, I have values of times in seconds that I am interested in to make the calculation of the percentage of the piece of pie, but I want that information to be formatted in HH:MM when passing over it. 

I know that in previous versions of primefaces it was possible to add to the Data a text and a value. Now it is only possible to add a numeric value. 

I don't think they have gone backwards in this aspect but I have already tried several things and I can't find a way to do what I am saying.

Thank you.

**Comment by melloware** [comment]
Don't you just want to customize the Tooltip of the Chart using the Charts.JS tooltip?  https://www.chartjs.org/docs/latest/configuration/tooltip.html

PrimeFaces is just sitting on top of ChartsJS.

**Comment by zapeSpawn** [comment, marked as answer]
I finally solved it by redefining the client-side tooltip:

<p:barChart model="#{infInformeClientes.stackedBarLlamadasTiempoConversadoModel}" style="width: 100%; height: 500px;" widgetVar="stackedBarLlamadasTiempoConversadoModelWgt"/>
                                            <script>                                                                                                                        
                                                PrimeFaces.widgets.stackedBarLlamadasTiempoConversadoModelWgt.cfg.config.options.plugins.tooltip = {
                                                    callbacks: {
                                                        label: function(context) {
                                                            let label = context.label || '';
                                                            if (label) {
                                                                label += ': ';
                                                            }
                                                            if (context.raw !== null) {                                                                                                
                                                                const horasMinutos = segundosAHorasYMinutos(context.raw);
                                                                label += (horasMinutos.horas + ":" + horasMinutos.minutos);
                                                            }
                                                            return label;
                                                        }
                                                    }
                                                };                                                                                                                                                                
                                                PrimeFaces.widgets.stackedBarLlamadasTiempoConversadoModelWgt.refresh(PrimeFaces.widgets.stackedBarLlamadasTiempoConversadoModelWgt.cfg);                                                                            
</script>
